dev-lang/clojure/clojure-1.10.0.ebuild

```diff
@@ -32,17 +29,653 @@ S="${WORKDIR}/clojure-clojure-76b87a6"
+	org.sonatype.nexus.plugins:nexus-plugins-restlet1x:pom:2.9.1-02
+	org.sonatype.nexus.plugins:nexus-plugins:pom:2.9.1-02
+	org.sonatype.nexus.plugins:nexus-restlet1x-model:jar:2.9.1-02
+	org.sonatype.nexus:nexus-client-core:jar:2.9.1-02
+	org.sonatype.nexus:nexus-components:pom:2.9.1-02
+	org.sonatype.nexus:nexus-oss:pom:2.9.1-02
+	org.sonatype.oss:oss-parent:pom:5
+	org.sonatype.oss:oss-parent:pom:7
+	org.sonatype.plexus:plexus-build-api:jar:0.0.4
+	org.sonatype.plexus:plexus-cipher:jar:1.4
+	org.sonatype.plexus:plexus-cipher:jar:1.7
+	org.sonatype.plexus:plexus-sec-dispatcher:jar:1.3
+	org.sonatype.plexus:plexus-sec-dispatcher:jar:1.4
+	org.sonatype.plugins:nexus-staging-maven-plugin:jar:1.6.7
+	org.sonatype.sisu.inject:guice-bean:pom:1.4.2
+	org.sonatype.sisu.inject:guice-plexus:pom:1.4.2
+	org.sonatype.sisu.siesta:siesta-client:jar:1.7
+	org.sonatype.sisu.siesta:siesta-common:jar:1.7
+	org.sonatype.sisu.siesta:siesta-jackson:jar:1.7
+	org.sonatype.sisu.siesta:siesta:pom:1.7
+	org.sonatype.sisu:sisu-guice:jar:noaop:2.1.7
+	org.sonatype.sisu:sisu-inject-bean:jar:1.4.2
+	org.sonatype.sisu:sisu-inject-plexus:jar:1.4.2
+	org.sonatype.sisu:sisu-inject:pom:1.4.2
+	org.sonatype.sisu:sisu-parent:pom:1.4.2
+	org.sonatype.spice.zapper:spice-zapper:jar:1.3
+	org.sonatype.spice:spice-parent:pom:10
+	org.sonatype.spice:spice-parent:pom:12
+	org.sonatype.spice:spice-parent:pom:15
+	org.sonatype.spice:spice-parent:pom:16
+	org.sonatype.spice:spice-parent:pom:17
+	org.vafer:jdependency:jar:1.2
+	oro:oro:jar:2.0.8
+	xmlpull:xmlpull:jar:1.1.3.1
+	xml-apis:xml-apis:jar:1.0.b2
+	xml-apis:xml-apis:jar:2.0.2
+	xpp3:xpp3_min:jar:1.1.4c
+)
+
+__extract_info() {
+	local f="$1" host="$2" artifact="$3"
+	local rest="${artifact}" group name type version classifier
+	group=${rest%%:*}
+	rest=${rest#*:}
+	name=${rest%%:*}
+	rest=${rest#*:}
+	type=${rest%%:*}
+	rest=${rest#*:}
+	version=${rest%%:*}
+
+	if ! [[ "${rest}" = "${rest#*:}" ]] ; then
+		classifier="${version}"
+		rest=${rest#*:}
+		version=${rest%%:*}
+	fi
+
+	${f} "${host}" "${group}" "${name}" "${type}" "${version}" "${classifier}" || die
+}
+
+__jar_and_pom() {
+	local f="$1" host="$2" group="$3" name="$4" type="$5" version="$6" classifier="$7"
+	${f} "${host}" "${group}" "${name}" "${type}" "${version}" "${classifier}" || die
+	# For JARs we also always need POMs:
+	if [[ "${type}" = jar ]] ; then
+		${f} "${host}" "${group}" "${name}" pom "${version}" "${classifier}" || die
+	fi
+}
+
+__add_to_src_uri() {
+	local host="$1" group="$2" name="$3" type="$4" version="$5" classifier="$6"
+	local path="${group//.//}"
+	local directory="${path}/${name}/${version}" file="${name}-${version}.${type}"
+	if [[ "${type}" != pom ]] && [[ -n "${classifier}" ]] ; then
+		file="${name}-${version}-${classifier}.${type}"
+	fi
+	SRC_URI+=" ${host}/${directory}/${file} -> ${group}:${file}"
+}
+
+__set_vendor_uri() {
+	local lib host artifact
+	for lib in "${EMAVEN_ARTIFACTS[@]}"; do
+		host=${lib%% *}
```

Comment:
Is this actually used?

Reply:
In my example '$host` is not used.  But `net-p2p/bisq-0.6.3` shows that generally there is a necessity to be able to specify other artifact repository hosts than Maven Central.  Since I would like to build an eclass from this, I opted to generalise the functionality.